pycue/opencue/wrappers/job.py

```diff
@@ -187,6 +187,17 @@ def getLayers(self):
         layerSeq = response.layers
         return [opencue.wrappers.layer.Layer(lyr) for lyr in layerSeq.layers]
 
+    def getLayer(self, layerName):
+        """ Returns the layer with the specified name
+        :type:   layername: str
+        :rtype:  opencue.wrappers.layer.Layer
+        :return: specific layer in the job
+        """
+        for layer in self.getLayers():
+            if layer.name() == layerName:
+                return layer
+        return None
```

Comment:
[api.py](https://github.com/AcademySoftwareFoundation/OpenCue/blob/master/pycue/opencue/api.py#L402) has a function to find a frame based on a job and layer name. I agree it is useful to have a `getLayer` function on the job wrapper, but maybe it should query directly the layer it needs using the same logic as api.py instead of loading all the layers into memory to search for a single one.

```python
@util.grpcExceptionParser
def findLayer(job, layer):
    """Finds and returns a layer from the specified pending job.

    :type job: str
    :param job: the job name
    :type layer: str
    :param layer: the layer name
    :rtype: opencue.wrappers.layer.Layer
    :return: the layer matching the query"""
    return Layer(Cuebot.getStub('layer').FindLayer(
        job_pb2.LayerFindLayerRequest(job=job, layer=layer), timeout=Cuebot.Timeout).layer)
```

Reply:
But wouldn't this require to ask the server? The information is already there, so don't see a need to ask the server. It should also be alot faster since it doesn't ask the server.

Reply:
Nevermind the previous comment. Just realized that the getLayers() request the layers from the server.

Reply:
Was able to optimize the request, so it only fetches the specific layer